Modules/Registration/Metricsv4/include/itkPointSetToPointSetMetricv4.hxx

```diff
@@ -342,7 +342,7 @@ PointSetToPointSetMetricv4<TFixedPointSet, TMovingPointSet, TInternalComputation
 
       if( this->m_CalculateValueAndDerivativeInTangentSpace )
         {
-        for( DimensionType d = 0; d < PointDimension; ++d )
+        for( DimensionType d = 0; d < this->GetNumberOfLocalParameters(); ++d )
```

Comment:
If these were different, that would always fail, not just sporadically.

Reply:
I thought the behavior is undefined if you write out of bounds? Depending on what you end up writing to it might or might not crash?

Reply:
I agree with @samuelgerber .  What is the size of "pointDerivative" that is indexed by 'd'?

Reply:
Writing out of bounds usually gets reported in debug mode. I will run this in debugger to double check.

Reply:
threadLocalTransformDerivative: size 2
pointDerivative: size 2
PointDimension: size 2

Reply:
Thanks @dzenanz, now I can stop barking up the wrong tree :)